chesster.js

```diff
@@ -62,268 +47,89 @@ function critical_path(todo){
-            var byId = {};
-            var total = response.channels.length;
-            for (var i = 0; i < total; i++) {
-                var channel = response.channels[i];
-                byName[channel.name] = channel;
-                byId[channel.id] = channel;
-            }
-            channels.byName = byName;
-            channels.byId = byId;
-        }
-        console.log("info: got channels");
+    function (bot, message){
+        return message.league[responseName]().then(function(response) {
+            bot.reply(message, response);
+        });
     });
-
 }
-
-/* 
-   updates the user list
-   updates the channel lists
-   then repeats in new thread
-   
-   if it encounters an error it will exit the process with exit code 1
-*/
-var count = 0;
-function refresh(bot, delay) {
-    critical_path(function(){
-        console.log("doing refresh " + count++);
-        bot.rtm.ping();
-        
-        update_users(bot);
-        update_channels(bot);
-        setTimeout(function(){ 
-            refresh(bot, delay);
-        }, delay);
+// A helper for a very common pattern
+function leagueDMResponse(patterns, responseName) {
+    chesster.hears({
+        middleware: [slack.requiresLeague],
+        patterns: patterns,
+        messageTypes: [
+            'direct_message',
+            'direct_mention'
+        ]
+    },
+    function (bot, message){
+        var deferred = Q.defer();
+        bot.startPrivateConversation(message, function (response, convo) {
+            message.league[responseName]().then(function(response) {
+                convo.say(response);
+                deferred.resolve();
+            }, function(error) {
+                deferred.reject(error);
+            });
+        });
+        return deferred.promise;
     });
 }
 
-controller.spawn({
-  token: config.token
-}).startRTM(function(err, bot) {
-
-    if (err) {
-        throw new Error(err);
-    }
-    
-    //refresh your user and channel list every 2 minutes
-    //would be nice if this was push model, not poll but oh well.
-    refresh(bot, 120 * SECONDS);
-
-});
-
 /* stop giritime */
-
-controller.hears([
-	'giritime'
-],[
-	'ambient'
-],function(bot,message) {
-    bot_exception_handler(bot, message, function(){
-        var response = "Stop... Giri Time!\n" + "Hi! Im Chesster. Ill be your new bot. " + 
-						"To interact with me, mention " + users.getIdString("chesster") + 
-						" in a message";
-        bot.reply(message, response);
-    });
+chesster.hears({
+    patterns: 'giritime',
+    messageTypes: [
+        'ambient'
+    ]
+},
+function(bot,message) {
+    var response = "Stop... Giri Time!\n" + "Hi! Im Chesster. Ill be your new bot. " + 
+                    "To interact with me, mention " + slack.users.getIdString("chesster") + 
+                    " in a message";
+    bot.reply(message, response);
 });
 
 /* captains */
+leagueResponse(['captain guidelines'], 'formatCaptainGuidelinesResponse');
```

Comment:
This requires a league? What does it do if raised in #general channel?

Ill probably find the answer to this later but just in case i dont, I am asking now.